pkg/controller/baremetalhost/baremetalhost_controller.go

```diff
@@ -198,6 +200,22 @@ func (r *ReconcileBareMetalHost) Reconcile(request reconcile.Request) (result re
 		}
 	}
 
+	// re-apply host status from annotation
+
+	content := []byte(annotations["kubectl.kubernetes.io/last-applied-configuration"])
+	// Convert the yaml into a typed object
+	obj := &metal3v1alpha1.BareMetalHost{}
+	codecFactory := serializer.NewCodecFactory(r.scheme)
+
+	if err := runtime.DecodeInto(codecFactory.UniversalDecoder(), content, obj); err != nil {
+		reqLogger.Info("failed to decode annotation %v", err)
+	}
+
+	// Manually restore data.
+	if !reflect.DeepEqual(host.Status, obj.Status) {
```

Comment:
This should be done only if the status is empty and the annotation is not empty

Reply:
I am not sure if I understood you correctly. what about when both are not empty and they have different values

Reply:
discard the annotation then. The annotation should only be used in case we loose the status IMO

Reply:
How will you make sure then that after pivoting status is grabbed from annotation automatically?

Reply:
Check if the status is empty. If so, and you have an annotation, you can use it to populate the status again. Populating the status should not be something done on regular basis.

Reply:
You are right, we must do it only when needed, I am rechecking the code.